### ISSUE TITLE
Publish the extension on Edge

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor-chrome-extension#184]
> Microsoft rolled out the next stage: It is now possible to submit an extension for review.

Request submitted today!
https://developer.microsoft.com/en-us/microsoft-edge/extensions/requests/

**Comment by Mogztter** [comment]
> For Microsoft Edge, all extension APIs are under the browser namespace, e.g. browser.browserAction.disable()

Currently the extension is using the `chrome` namespace.
If all browsers are using the (new) `browser` namespace then we should use it. If only Edge is using the new `browser` namespace then we should test if `chrome` namespace is defined and if not use the `browser` namespace.

https://docs.microsoft.com/en-us/microsoft-edge/extensions/api-support/supported-apis

> Request submitted today!

Actually a few steps are missing... I need to follow this guide to package the extension: https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/packaging/creating-and-testing-extension-packages

**Comment by Mogztter** [comment]
The extension is now working on Edge 

![edge](https://user-images.githubusercontent.com/333276/33527175-494efb54-d84c-11e7-8380-a94c5af000ff.png)

I will publish an unsigned package to GitHub soon.
Hopefully the next version will be signed once I've figured out how to sign a package!

**Comment by mojavelinux** [comment]
:tada:

**Comment by Mogztter** [comment]
I think we should use Manifoldjs to generate the vendor specific archive: https://docs.microsoft.com/en-us/microsoft-edge/extensions/guides/packaging/using-manifoldjs-to-package-extensions

**Comment by Mogztter** [comment]
We will wait until the next version of Edge (based on Chromium) is publicly available.

**Comment by Mogztter** [comment]
You can install extensions from other stores in Edge (Chromium).
That means that you can install the Asciidoctor browser extension from the Chrome Web Store.

1. Go to `edge://extensions`
2. Check Allow extensions from other stores
3. Go to https://chrome.google.com/webstore/detail/asciidoctorjs-live-previe/iaalpfgpbocpdfblpnhhgllgbdbchmia
4. Install the extension

As far as I know the Microsoft store for Edge extensions is still in closed beta (invitation only) so I cannot publish the extension (yet).

**Comment by sysfu** [comment]
Any update on this with regards to Edge Dev? I'm running Edge Dev 83.0.478.5 (Official build) dev (64-bit) and just updated to the 2.5.0 release, but the extension seems to remain broken on this platform going on several months.

**Comment by Mogztter** [comment]
I didn't know... thanks for your report! I will check on my machine.
Do you see any error in the console? Is it 

It's surprising because Edge is now using Chromium and the last time I checked the Web Extensions API was almost identical to Chrome... 🤔

**Comment by Mogztter** [comment]
@sysfu seems to be working fine on my machine with Edge Dev 83.0.478.10 and version 2.5.0:

### Extension installed
![edge-ext-installed](https://user-images.githubusercontent.com/333276/79617221-3c9e8f80-8107-11ea-89c8-22c590f07cc1.PNG)


### Local file
![ok-local-edge](https://user-images.githubusercontent.com/333276/79617222-3e685300-8107-11ea-8e89-9c523fdd29ea.png)


### Remove file (GitHub)
![remote-ok-edge](https://user-images.githubusercontent.com/333276/79617228-40caad00-8107-11ea-98bb-83b023bee38d.png)


> but the extension seems to remain broken on this platform going on several months.

Could you please clarify what is not working exactly?

**Comment by sysfu** [comment]
I think I discovered the issue; it does not work with documents that are loaded from a UNC network path. Should I open a new issue for this?

**Comment by Mogztter** [comment]
Yes please, this issue is more about publishing the extension to the Edge store (but since it can be installed through the Chrome extension store I didn't put too much focus on it).

**Comment by sysfu** [comment]
Should https://github.com/asciidoctor/asciidoctor-browser-extension/issues/94 be re-opened, or do you think this merits a new issue?

**Comment by Mogztter** [comment]
Can you confirm that this is the same issue? Do you have a CORS error? If that is the case then the only way to work around it is to do one of:

- Setup a webserver
- Copy the file to your local disk
- Disable the web security on Edge Dev (not sure if the `--disable-web-security` option is working on Edge Dev)

**Comment by sysfu** [comment]
I don't think it's CORS, opening a new issue

**Comment by Mogztter** [comment]
@sysfu I had an idea to render the document. I think we can get the initial content from the current page/document (without using an `XMLHttpRequest`).
Having said that, this solution will only work if you disable the live reload so, if you update your AsciiDoc document, changes won't be visible unless you manually refresh the page. Do you think it's worth it?

**Comment by Mogztter** [comment]
Sorry I meant to comment on https://github.com/asciidoctor/asciidoctor-browser-extension/issues/353 :sweat_smile:

**Comment by sysfu** [comment]
> @sysfu I had an idea to render the document. I think we can get the initial content from the current page/document (without using an `XMLHttpRequest`). Having said that, this solution will only work if you disable the live reload so, if you update your AsciiDoc document, changes won't be visible unless you manually refresh the page. Do you think it's worth it?

Just noticing this comment three years later. I think it would be worth it yes, but I have since moved on to another job where I don't make use of AsciiDoc. Refreshing a page is much less complicated than setting up a local web server or file sync.

**Comment by ggrossetie** [comment]
The extension has been published on the Microsoft Edge Add-ons store for a while now: https://microsoftedge.microsoft.com/addons/detail/asciidoctorjs-live-previ/pefkelkanablhjdekgdahplkccnbdggd (also linked from the README).

Closing this one since the original scope (publishing on Edge) is done. If you're running into a problem specific to Edge, please open a new issue with the details (Edge version, steps to reproduce, etc.) so we can track it separately.
